force-app/main/default/lwc/serviceScheduleCreator/serviceScheduleCreator.js

```diff
@@ -110,27 +137,40 @@ export default class ServiceScheduleCreator extends NavigationMixin(LightningEle
     handleNext() {
         if (this.isStep1) {
             this.processNewServiceSchedule();
-            // Set next will only happen in the processNewServiceSchedule if the record is valid and we can proceed.
         } else if (this.isStep2) {
             this.processSessions();
         } else if (this.isStep3) {
-            this.setNextStep();
+            this.processServiceParticipants();
         } else if (this.isStep4) {
-            this.save();
+            this.save(true);
         }
     }
 
-    save() {
+    save(restart) {
```

Comment:
nitpicky -- can we call the param `doRestart` ? i find this confusing without tracing it back.